src/Social.js

```diff
@@ -0,0 +1,83 @@
+import React, {Component, Children, createElement} from 'react'
+import classNames from 'classnames'
+import Button from "./Button"
+
+export default class Social extends Component {
+  static get propTypes() {
+    return {
+      socialNetwork: React.PropTypes.string,
+      url: React.PropTypes.string,
+      message: React.PropTypes.string,
+      button: React.PropTypes.bool
+    }
+  }
+  static get defaultProps() {
+    return {
+        socialNetwork: "Facebook",
+        url: "http://revelry.co",
+        message: "Check this out!",
+        button: false
+    }
+  }
+
+  getLink() {
+    const {url, message} = this.props
+    const encodedMsg = encodeURIComponent(message)
+
+    switch (this.props.socialNetwork) {
+      case "Buffer":
+        return `https://buffer.com/add?text=${encodedMsg}&url=${url}`
+      case "Digg":
+        return `http://digg.com/submit?url=${url}&title=${encodedMsg}`
+      case "email":
+        return `mailto:?subject=${encodedMsg}&body=${url}`
+      case "Facebook":
+        return `https://www.facebook.com/sharer/sharer.php?u=${url}`;
+      case "Google+":
+        return `https://plus.google.com/share?url=${url}`
+      case "LinkedIn":
+        return `https://www.linkedin.com/shareArticle?url=${url}&title=${encodedMsg}`
+      case "Pinterest":
+        return `https://pinterest.com/pin/create/bookmarklet/?url=${url}&description=${encodedMsg}`
+      case "Reddit":
+        return `https://reddit.com/submit?url=${url}&title=${encodedMsg}`
+      case "Tumblr":
+        return `https://www.tumblr.com/widgets/share/tool?canonicalUrl=${url}&caption=${encodedMsg}`
+      case "Twitter":
+        return `https://twitter.com/intent/tweet?url=${url}&text=${encodedMsg}`
+      default:
+        console.warn("No valid social network given!")
+    }
+  }
```

Comment:
What if you define this above the component definition...

```
const URL_FUNCTIONS = {
  EMAIL: (url, text) => `mailto:?subject=${text}&body=${url}`,
  FACEBOOK: (url) => `https://www.facebook.com/sharer/sharer.php?u=${url}`,
  GOOGLE_PLUS: (url) => `https://plus.google.com/share?url=${url}`,
  ...
}
```

Then this in the component:

```
static types = URL_FUNCTIONS;

get href() {
  const {type, url, message} = this.props
  
  return type(encodeURIComponent(url), encodeURIComponent(message))
}
```

And then usage looked like this:

```
<Social type={Social.types.FACEBOOK} url="foo" message="bar">Facebook</Social>
```

That would eliminate the magic strings and give the end developer an escape hatch for using a custom sharing type. (Oh, btw `type` would be defined as a required `func` prop.)